flat_mutation_reader.hh

```diff
@@ -373,7 +388,7 @@ public:
         requires FlattenedConsumer<Consumer>()
     )
     auto consume_in_thread(Consumer consumer, db::timeout_clock::time_point timeout) {
-        return consume_in_thread(std::move(consumer), [] (const dht::decorated_key&) { return true; }, timeout);
+        return consume_in_thread(std::move(consumer), filter(), timeout);
```

Comment:
Would this mean we'll be paying for virtual dispatch inside `filter::partition_filer::operator()`? Perhaps we have a `no_filter` object with non-virtual methods which can be fully inlined.

Reply:
Not exactly, just calling indirectly via a pointer.  But we avoid this as well with a `no_filter` object as you suggested.